tests/components/sharkiq/test_config_flow.py

```diff
@@ -0,0 +1,138 @@
+
+    with patch(
+        "homeassistant.components.sharkiq.config_flow.get_ayla_api",
+        return_value=mocked_ayla,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG,
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    mocked_ayla = _create_mocked_ayla(connect=aiohttp.ClientError)
+
+    with patch(
+        "homeassistant.components.sharkiq.config_flow.get_ayla_api",
+        return_value=mocked_ayla,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG,
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_other_error(hass):
+    """Test we handle other errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    mocked_ayla = _create_mocked_ayla(connect=TypeError)
+
+    with patch(
+        "homeassistant.components.sharkiq.config_flow.get_ayla_api",
+        return_value=mocked_ayla,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_reauth(hass):
+    """Test reauth flow."""
+    with patch(
+        "homeassistant.components.sharkiq.vacuum.async_setup_entry", return_value=True,
+    ), patch("sharkiqpy.AylaApi.async_sign_in", return_value=True):
+        mock_config = MockConfigEntry(domain=DOMAIN, unique_id=UNIQUE_ID, data=CONFIG)
+        mock_config.add_to_hass(hass)
+        hass.config_entries.async_update_entry(mock_config, data=CONFIG)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "reauth", "unique_id": UNIQUE_ID}, data=CONFIG
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "reauth_successful"
+
+    with patch("sharkiqpy.AylaApi.async_sign_in", side_effect=SharkIqAuthError):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "reauth", "unique_id": UNIQUE_ID}, data=CONFIG,
+        )
+        assert result["type"] == "form"
+        assert result["errors"] == {"base": "invalid_auth"}
+
+    with patch("sharkiqpy.AylaApi.async_sign_in", side_effect=RuntimeError):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "reauth", "unique_id": UNIQUE_ID}, data=CONFIG,
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "unknown"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)